libs/common-types/src/lib.rs

```diff
@@ -53,6 +56,45 @@ pub enum PoolRole<TrancheId = [u8; 16], Moment = u64> {
 	TrancheInvestor(TrancheId, Moment),
 }
 
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, TypeInfo, Debug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum PermissionedCurrencyRole<Moment = u64> {
+	/// This role can hold & transfer tokens
+	Holder(Moment),
+	/// This role can add/remove holders
+	Manager,
+	/// This role can mint/burn tokens
+	Issuer,
+}
+
+/// The Role enum is used by the permissions pallet,
+/// to specify which role an account has within a
+/// specific scope.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, TypeInfo, Debug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Role<TrancheId = [u8; 16], Moment = u64> {
+	/// Roles that apply to a specific pool.
+	PoolRole(PoolRole<TrancheId, Moment>),
+	/// Roles that apply to a specific permissioned currency.
+	PermissionedCurrencyRole(PermissionedCurrencyRole<Moment>),
+}
+
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, TypeInfo, Debug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum PermissionScope<PoolId, CurrencyId> {
+	Pool(PoolId),
+	Currency(CurrencyId),
+}
+
+impl<PoolId, CurrencyId> Default for PermissionScope<PoolId, CurrencyId>
+where
+	PoolId: Default,
+{
+	fn default() -> Self {
+		Self::Pool(PoolId::default())
+	}
+}
```

Comment:
Can we hide this behind an `#[cfg(test)]` guard?

Reply:
Good point 👍